### ISSUE TITLE
RSA sign 3b: TLS 1.2 integration testing

### ISSUE THREAD

**Issue by mpg** [Mbed-TLS/mbedtls#5628]
Test use of `MBEDTLS_PK_OPAQUE` RSA keys in the TLS library for TLS 1.2.

- [x] Test use for client authentication, similarly to what's done for opaque ECDSA keys, see `key_opaque` option in `ssl_client2.c` test cases using it in `ssl-opt.sh`. It may be enough to just add new cases to `ssl-opt.sh` with an RSA key (and matching cert) without needing to modify `ssl_client2.c`.
- [x] Test use for server authentication, similarly to what's done for opaque ECDSA keys, see `key_opaque` option in `ssl_server2.c` test cases using it in `ssl-opt.sh`. It may be enough to just add new cases to `ssl-opt.sh` with an RSA key (and matching cert) without needing to modify `ssl_server2.c`. **Note:** the RSA key should be tested with both an ECDHE-RSA and DHE-RSA key exchange, using `force_ciphersuite`. (Other key exchanges with "RSA" in their name use decryption and are out of scope, they will be handled in another series of tasks.)

Depends on: #5626 for `mbedtls_pk_wrap_opaque()` to work with RSA.
